src/pretix/plugins/paypal/payment.py

```diff
@@ -218,114 +276,168 @@ def settings_content_render(self, request):
 
         return settings_content
 
+    def get_isu_referral_url(self, request):
+        pprov = PaypalMethod(request.event)
+        request.session['payment_paypal_isu_event'] = request.event.pk
+        request.session['payment_paypal_isu_tracking_id'] = get_random_string(length=127)
+
+        pprov.init_api()
+
+        req = PartnerReferralCreateRequest()
+
+        req.request_body({
+            "operations": [
+                {
+                    "operation": "API_INTEGRATION",
+                    "api_integration_preference": {
+                        "rest_api_integration": {
+                            "integration_method": "PAYPAL",
+                            "integration_type": "THIRD_PARTY",
+                            "third_party_details": {
+                                "features": [
+                                    "PAYMENT",
+                                    "REFUND",
+                                    "ADVANCED_TRANSACTIONS_SEARCH",
+                                    "TRACKING_SHIPMENT_READWRITE",
+                                    "ACCESS_MERCHANT_INFORMATION"
+                                ],
+                            }
+                        }
+                    }
+                }
+            ],
+            "products": [
+                "EXPRESS_CHECKOUT"
+            ],
+            "partner_config_override": {
+                "partner_logo_url": "https://pretix.eu/static/pretixbase/img/pretix-logo.svg",
+                "return_url": build_global_uri('plugins:paypal:oauth.return')
+            },
+            "tracking_id": request.session['payment_paypal_isu_tracking_id'],
+            "preferred_language_code": request.user.locale
+        })
+
+        try:
+            response = pprov.client.execute(req)
+        except IOError as ioe:
+            if isinstance(ioe, HttpError):
+                messages.error(request, _('An error occurred during connecting with PayPal, please try again.'))
+        else:
+            return self.get_link(response.result.links, 'action_url').href
+
+    def get_link(self, links, rel):
+        for link in links:
+            if link.rel == rel:
+                return link
+
+        return None
+
+
+class PaypalMethod(BasePaymentProvider):
```

Comment:
Can `matching_id` and `api_payment_details` really stay the same if the info_data format changes a lot?